solutions/kayenta/ci/scripts/automated-canary.sh

```diff
@@ -0,0 +1,159 @@
+    -o=jsonpath='{.items[0].metadata.name}')
+kubectl -n spinnaker exec $HALYARD_POD -- sh -c "hal config canary google enable"
+kubectl -n spinnaker exec $HALYARD_POD -- sh -c "hal config canary google account add kayenta-tutorial --project $GOOGLE_CLOUD_PROJECT"
+kubectl -n spinnaker exec $HALYARD_POD -- sh -c "hal config canary google edit --stackdriver-enabled=true"
+kubectl -n spinnaker exec $HALYARD_POD -- sh -c "hal deploy apply"
+
+set +x
+timeout=0
+while [ "$timeout" -lt 600 -a $(kubectl -n spinnaker get pods | grep "kayenta" | wc -l) -ne 1 ]; do
+  echo "[${timeout}s] Spinnaker not yet ready, waiting 15s more."
+  sleep 15
+  timeout=$((timeout+15))
+done
+if [ "$timeout" -ge 600 ]; then
+  echo "Timeout configuring Kayenta"
+  exit 1
+fi
+echo "Spinnaker ready!"
+set -x
+
+## Enabling canary on sampleapp ##
+DECK_POD=$(kubectl -n spinnaker get pods -l \
+    cluster=spin-deck,app=spin \
+    -o=jsonpath='{.items[0].metadata.name}')
+kubectl -n spinnaker port-forward $DECK_POD 8080:9000 \
+    >/dev/null &
+sleep 5
+
+curl -XPOST -d '{"job":[{"type":"updateApplication","application":{"name":"sampleapp","dataSources":{"enabled":["canaryConfigs"],"disabled":[]}},"user":"[anonymous]"}],"application":"sampleapp","description":"Update Application: sampleapp"}' \
+    --fail -sS -H "Content-Type: application/json" -H "Accept: */*" \
+    http://localhost:8080/gate/applications/sampleapp/tasks >/dev/null
+sleep 5
+
+## Creating Canary config for sampleapp ##
+curl -XPOST -d '{"name":"kayenta-test","applications":["sampleapp"],"description":"","metrics":[{"analysisConfigurations":{"canary":{"direction":"increase"}},"name":"error_rate","query":{"type":"stackdriver","serviceType":"stackdriver","customFilterTemplate":"http_code","metricType":"external.googleapis.com/prometheus/requests"},"groups":["Group 1"],"scopeName":"default"}],"configVersion":"1","templates":{"http_code":"metric.labels.http_code = \"500\" AND resource.label.pod_name = starts_with(\"${scope}\")"},"classifier":{"groupWeights":{"Group 1":100},"scoreThresholds":{"pass":95,"marginal":75}},"judge":{"name":"NetflixACAJudge-v1.0","judgeConfigurations":{}}}' \
+    --fail -sS -H "Content-Type: application/json" -H "Accept: */*" \
+    http://localhost:8080/gate/v2/canaryConfig > /dev/null
+sleep 5
+
+## Create automated canary pipeline ##
+export PIPELINE_ID=$(curl localhost:8080/gate/applications/sampleapp/pipelineConfigs/Simple%20deploy \
+    | jq -r '.id')
+export CANARY_CONFIG_ID=$(curl localhost:8080/gate/v2/canaryConfig | jq -r '.[0].id')
+jq '(.stages[] | select(.refId == "9") | .pipeline) |= env.PIPELINE_ID | (.stages[] | select(.refId == "8") | .pipeline) |= env.PIPELINE_ID | (.stages[] | select(.refId == "11") | .canaryConfig.canaryConfigId) |= env.CANARY_CONFIG_ID' spinnaker-git/solutions/kayenta/pipelines/automated-canary.json | \
+    curl -d@- -X POST \
+    -H "Content-Type: application/json" -H "Accept: */*" \
+    http://localhost:8080/gate/pipelines
```

Comment:
@jtk54 You have both examples on this one.

1. The pipeline that is created here depends on another pipeline (it fetches an artifact and also can trigger it). I need to inject the ID of that other pipeline in the definition of this new pipeline.
2. This new pipeline uses a Canary Analysis stage, and that takes a canary config as parameter. I need to inject the ID of this canary config in the definition of this new pipeline.

Reply:
What's preventing you from generating the desired pipeline spec given that you know the pipeline & canary config IDs before submitting it with spin?

Reply:
Nothing, now that you mention it 😄However, I would expect a tool like spin to completely replace my need for curl (which I need to get the IDs). When writing the tutorial, I ruled against it for this reason: mixing curl and spin was super confusing. I just didn't think too much about it when writing the tests.

Reply:
The curl that was necessary was specifically for canary configs, right? That seems like a reasonable set of commands to add into `spin` going forward.

Reply:
Seems like we could have a `spin pipeline get-id` command as a companion to this one: `spin pipeline get --name my-pipeline --application my-app`.
And a similar thing for the as-yet-to-exist `canary-config` command group.

Reply:
Or just args to tweak the response content/format.

Reply:
What is wrong with using `jq` to parse the output coming from `spin` in the case where you want the pipeline id? Agreed that pipeline templates need a surface, but we need to think very carefully what that looks like given we are starting the design of managed pipeline templates v2.

Reply:
> What is wrong with using jq to parse the output coming from spin in the case where you want the pipeline id

I can see that a dependency on `jq` is not ideal, but on the other hand providing _some_ amount of response parsing/control in `spin` itself risks getting into a "feature race" with tools like `jq`, which is not the intention of `spin`.

If sufficiently common use cases show up that merit out-of-the-box support I'd be up for @duftler's suggestion of adding commands like `get-id`, but I think we'd need to wait a bit before figuring out what those use cases might be?

Reply:
I'm not sure I'm following the conversation between here and spinnaker/spin#59, but for what it's worth: I agree that templating is probably something that should be considered very carefully.
However, regarding the output of spin, it seems to me that a `--json-path` flag is not that complicated to implement using existing libs (as it seems it has been done on spinnaker/spin#59). I don't have any hard data, but I see a lot of people using this option with kubectl. As long as you don't use it to _change_ the content of response, just its formatting, I don't think there's a risk of "feature race" with jq. And if people want to use jq, they still can.
To me it looks like a quick win that improves the usability of spin.